tests/integration/conftest.py

```diff
@@ -689,41 +787,52 @@ def kernel(shell):
-            repo_name = "alma_old"
-            _create_old_repo(distro="alma", repo_name=repo_name)
-            if "8.8" in SYSTEM_RELEASE_ENV:
-                assert shell(f"yum install kernel-4.18.0-372.13.1.el8_6.x86_64 -y --enablerepo {repo_name}")
-                shell("grub2-set-default 'AlmaLinux (4.18.0-372.13.1.el8_6.x86_64) 8.6 (Sky Tiger)'")
-            else:
-                assert shell(f"yum install kernel-4.18.0-477.27.2.el8_8.x86_64 -y --enablerepo {repo_name}")
-                shell("grub2-set-default 'AlmaLinux (4.18.0-477.27.2.el8_8.x86_64) 8.8 (Sapphire Caracal)'")
-        elif "rocky-8" in SYSTEM_RELEASE_ENV:
-            repo_name = "rocky_old"
-            _create_old_repo(distro="rocky", repo_name=repo_name)
-            if "8.8" in SYSTEM_RELEASE_ENV:
-                assert shell(f"yum install kernel-4.18.0-372.13.1.el8_6.x86_64 -y --enablerepo {repo_name}")
-                shell("grub2-set-default 'Rocky Linux (4.18.0-372.13.1.el8_6.x86_64) 8.6 (Green Obsidian)'")
-            else:
-                assert shell(f"yum install kernel-4.18.0-477.27.1.el8_8.x86_64 -y --enablerepo {repo_name}")
-                shell("grub2-set-default 'Rocky Linux (4.18.0-477.27.1.el8_8.x86_64) 8.8 (Green Obsidian)'")
-        elif "stream-8" in SYSTEM_RELEASE_ENV:
-            # Given the CentOS Stream rolling release, we are able to just downgrade the kernel
-            assert shell("yum downgrade kernel -y").returncode == 0
+        # Verify that there is multiple kernels installed
+        if int(shell("rpm -q kernel | wc -l").output.strip()) > 1:
+            # We don't need to do anything at this point
+            # The whole setup needed happens after
+            pass
+
+        # There won't be much changes for EL 7 packages anymore
+        # We can hardcode this then
+        # The release part differs a bit on CentOS and Oracle,
+        # so going with wildcard asterisk to generalize
+        elif SystemInformationRelease.version.major == 7:
+            older_kernel = "kernel-3.10.0-1160.118*"
+            assert shell(f"yum install -y {older_kernel}").returncode == 0
+
+        # Try to downgrade kernel version, if there is not multiple versions installed already.
+        # If the kernel downgrade fails, assume it's not possible and try to install from
+        # an older repo. This should only happen when Alma and Rocky has just landed on
+        # a fresh minor version.
+        elif shell("yum downgrade kernel -y").returncode != 0:
+            # Assuming this can only happen with Alma and Rocky we'll try to install an older kernel
+            # from a previous minor version.
+            # For that we need to use the vault url and bump te current minor down one version.
+            major_ver = SystemInformationRelease.version.major
+            minor_ver = SystemInformationRelease.version.minor
+            previous_minor_ver = minor_ver - 1
+            if minor_ver == 0:
```

Comment:
doesn't hurt to put something like that
```suggestion
            if minor_ver <= 0:
```

Reply:
Doesn't hurt, but doesn't make sense either.
We can't ever land on a system with `minor_ver < 0`
But yeah, I can include that, I guess :shrug:

Reply:
Oh yeah, you are right, I thought for some reason that you are using `previous_minor_ver` in that condition where you are subtracting -1  which could lead to negative number. The `minor_ver` is fine